fe/fe-core/src/main/java/org/apache/doris/common/util/RuntimeProfile.java

```diff
@@ -165,18 +173,23 @@ public void update(final TRuntimeProfileTree thriftProfile) {
     // preorder traversal, idx should be modified in the traversal process
     private void update(List<TRuntimeProfileNode> nodes, Reference<Integer> idx) {
         TRuntimeProfileNode node = nodes.get(idx.getRef());
-        // Make sure to update the latest LoadChannel profile according to the timestamp.
+        // Make sure to update the latest LoadChannel profile according to the
+        // timestamp.
         if (node.timestamp != -1 && node.timestamp < timestamp) {
             return;
         }
+        if (node.isSetMetadata()) {
+            this.nodeid = (int) node.getMetadata();
```

Comment:
I think we should not depend on it.
Add a specific field in profile object named plan_node_id. 
metadata is too normal.